local/gugcat/classes/grade_aggregation.php

```diff
@@ -627,11 +627,12 @@ public static function adjust_course_weight($weights, $courseid, $studentid, $no
      * Release final assessment grades for all the students
      *
      * @param int $courseid
+     * @param int $categoryid
      */
-    public static function release_final_grades($courseid){
+    public static function release_final_grades($courseid, $categoryid){
         global $USER, $DB;
         //Retrieve modules and enrolled students per grade category
-        $modules = local_gugcat::get_activities($courseid);
```

Comment:
Just put `$categoryid = optional_param('categoryid', null, PARAM_INT);` inside this function, instead of adding a new parameter.